helix-term/src/ui/terminal.rs

```diff
@@ -0,0 +1,121 @@
+                    c.set_style(
+                        helix_view::theme::Style::reset()
+                            .bg(map_color(cell.bg))
+                            .fg(map_color(cell.fg)),
+                    );
+                }
+            }
+        }
+    }
+
+    pub(crate) fn handle_event(
+        &self,
+        event: &input::Event,
+        context: &mut Context,
+    ) -> Option<EventResult> {
+        if context.editor.terminals.visible {
+            match event {
+                input::Event::Key(input::KeyEvent {
+                    code: input::KeyCode::Char('\\'),
+                    ..
+                }) => {
+                    if self.prev_pressed.get() {
+                        context.editor.terminals.visible = false;
+                        Some(EventResult::Consumed(None))
+                    } else {
+                        self.prev_pressed.set(true);
+
+                        if context.editor.terminals.handle_input_event(event) {
+                            Some(EventResult::Consumed(None))
+                        } else {
+                            Some(EventResult::Ignored(None))
+                        }
+                    }
+                }
+
+                input::Event::Resize(_, _) => Some(EventResult::Ignored(None)),
+
+                event => {
+                    if let input::Event::Key(..) = &event {
+                        self.prev_pressed.set(false);
+                    }
+
+                    if context.editor.terminals.handle_input_event(event) {
+                        Some(EventResult::Consumed(None))
+                    } else {
+                        Some(EventResult::Ignored(None))
+                    }
+                }
+            }
+        } else {
+            None
+        }
+    }
+}
+
+fn map_color(col: alacritty_terminal::ansi::Color) -> Color {
+    match col {
+        alacritty_terminal::ansi::Color::Named(named) => match named {
+            NamedColor::Black => Color::Black,
+            NamedColor::Red => Color::Red,
+            NamedColor::Green => Color::Green,
+            NamedColor::Yellow => Color::Yellow,
+            NamedColor::Blue => Color::Blue,
+            NamedColor::Magenta => Color::Magenta,
+            NamedColor::Cyan => Color::Cyan,
+            NamedColor::White => Color::White,
+            NamedColor::BrightBlack => Color::Gray,
+            NamedColor::BrightRed => Color::LightRed,
+            NamedColor::BrightGreen => Color::LightGreen,
+            NamedColor::BrightYellow => Color::LightYellow,
+            NamedColor::BrightBlue => Color::LightBlue,
+            NamedColor::BrightMagenta => Color::LightMagenta,
+            NamedColor::BrightCyan => Color::LightCyan,
+            _ => Color::Reset,
+        },
+        alacritty_terminal::ansi::Color::Spec(c) => Color::Rgb(c.r, c.g, c.b),
+        alacritty_terminal::ansi::Color::Indexed(idx) => Color::Indexed(idx),
+    }
+}
```

Comment:
This should be in the `helix-view` crate, you can implement `From<alacritty_terminal::ansi::Color>` for `Color`